lib/tapioca/commands/dsl.rb

```diff
@@ -378,7 +378,12 @@ def generate_command_for(constant)
 
       sig { void }
       def load_dsl_extensions
-        Dir["#{__dir__}/../dsl/extensions/*.rb"].sort.each { |f| require(f) }
+        Dir.glob([
+          "#{__dir__}/../dsl/extensions/*.rb",
+          "#{@tapioca_path}/dsl_extensions/**/*.rb",
```

Comment:
I'm ok with the approach but the name bothers me as it's not matching what we do for `compilers`: https://github.com/Shopify/tapioca/blob/1ae637362417034cf4007e37758fe5fe1263f086/lib/tapioca/commands/dsl.rb#L158-L159

@paracycle should we move the compilers under `sorbet/tapioca/dsl/compilers` so extensions can be under `sorbet/tapioca/dsl/extensions`?

Reply:
With the new loader API, all of this will be gone, tbh, but yeah, I think it is better to have this be:
```suggestion
          "#{@tapioca_path}/dsl/extensions/**/*.rb",
```
instead.

Reply:
I wanted to go with just `extensions` but i thought that may be confusing.

@paracycle i remember seeing talk of the new loader API. is there anywhere I can get more info on it? I've made it work without this patch in my project, so can definitely hold out until the new API drops.